dev/NavigationView/NavigationView.cpp

```diff
@@ -1779,10 +1779,90 @@ void NavigationView::OnNavigationViewItemKeyDown(const winrt::IInspectable& send
     }
 }
 
+void NavigationView::ArrowKeyNavigationPolyfill(const winrt::NavigationViewItem& nvi, const winrt::KeyRoutedEventArgs& args)
+{
+    switch (args.Key())
+    {
+    case winrt::VirtualKey::Down:
+        if (auto const ir = GetParentItemsRepeaterForContainer(nvi))
+        {
+            if (ir != m_topNavRepeater.get())
+            {
+                auto const indexOfCurrentElement = GetIndexFromItem(ir, nvi);
+                if (indexOfCurrentElement >= 0)
+                {
+                    if (FocusNextFocusableElement(ir, indexOfCurrentElement))
+                    {
+                        args.Handled(true);
+                    }
+                }
+            }
+        }
+        break;
+    case winrt::VirtualKey::Up:
+        if (auto const ir = GetParentItemsRepeaterForContainer(nvi))
+        {
+            if (ir != m_topNavRepeater.get())
+            {
+                auto const indexOfCurrentElement = GetIndexFromItem(ir, nvi);
+                if (indexOfCurrentElement >= 0)
+                {
+                    if (FocusPreviousFocusableElement(ir, indexOfCurrentElement))
+                    {
+                        args.Handled(true);
+                    }
+                }
+            }
+        }
+        break;
+    }
+}
+
+bool NavigationView::FocusNextFocusableElement(const winrt::ItemsRepeater& ir, const int elementIndex)
+{
+    if (auto const itemsSourceView = ir.ItemsSourceView())
+    {
+        for (int i = elementIndex + 1; i < itemsSourceView.Count(); i++)
+        {
+            if (auto const nextElement = ir.TryGetElement(i).try_as<winrt::NavigationViewItem>())
+            {
+                SetFocus(nextElement, winrt::FocusState::Keyboard);
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+bool NavigationView::FocusPreviousFocusableElement(const winrt::ItemsRepeater& ir, const int elementIndex)
+{
+    if (auto const itemsSourceView = ir.ItemsSourceView())
+    {
+        for (int i = elementIndex - 1; i >= 0; i--)
+        {
+            if (auto const nextElement = ir.TryGetElement(i).try_as<winrt::NavigationViewItem>())
+            {
+                SetFocus(nextElement, winrt::FocusState::Keyboard);
```

Comment:
>SetFocus [](start = 16, length = 8)

What if SetFocus fails, for instance because the next NVI is disabled. I think you should only return here if SetFocus returns true, otherwise continue your for loop.